.github/workflows/pr_common_unused_deps.yaml

```diff
@@ -0,0 +1,35 @@
+name: Unused Dependencies
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["**"]
```

Comment:
running this on every pr seems overkill, maybe we can run every week or so and post to slack?